EECS3311-PD1/src/main/java/UI/BottomMenuView.java

```diff
@@ -0,0 +1,195 @@
+        private String chartName;
+
+        public SelectedChartEvent(String chartName, EventType<? extends Event> eventType) {
+            super(eventType);
+            this.chartName = chartName;
+        }
+
+        public String getChartName() {
+            return chartName;
+        }
+    }
+
+    static public class RecalculateEvent extends Event {
+        private String analysisName;
+        private String chartName;
+
+        public RecalculateEvent(String analysisName, String chartName, EventType<? extends Event> eventType) {
+            super(eventType);
+            this.analysisName = analysisName;
+            this.chartName = chartName;
+        }
+
+        public String getAnalysisName() {
+            return analysisName;
+        }
+
+        public void setAnalysisName(String analysisName) {
+            this.analysisName = analysisName;
+        }
+
+        public String getChartName() {
+            return chartName;
+        }
+
+        public void setChartName(String chartName) {
+            this.chartName = chartName;
+        }
+    }
+
+    //Object Property for the chart added button.
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////
+    private ObjectProperty<EventHandler<SelectedChartEvent>> propertyOnChartAdded = new SimpleObjectProperty<>();
+
+    public final ObjectProperty<EventHandler<SelectedChartEvent>> onChartAddedProperty() {
+        return propertyOnChartAdded;
+    }
+
+    public final void setOnChartAdded(EventHandler<SelectedChartEvent> handler) {
+        propertyOnChartAdded.set(handler);
+    }
+
+    public final EventHandler<SelectedChartEvent> getOnChartAdded() {
+        return propertyOnChartAdded.get();
+    }
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+
+    //Object Property for the chart removed button.
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////
+    private ObjectProperty<EventHandler<SelectedChartEvent>> propertyOnChartRemoved = new SimpleObjectProperty<>();
+
+    public final ObjectProperty<EventHandler<SelectedChartEvent>> onChartRemovedProperty() {
+        return propertyOnChartRemoved;
+    }
+
+    public final void setOnChartRemoved(EventHandler<SelectedChartEvent> handler) {
+        propertyOnChartRemoved.set(handler);
+    }
+
+    public final EventHandler<SelectedChartEvent> getOnChartRemoved() {
+        return propertyOnChartRemoved.get();
+    }
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+
+    //Object Property for the recalculate button.
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////
+    private ObjectProperty<EventHandler<RecalculateEvent>> propertyOnRecalculatePressed = new SimpleObjectProperty<>();
```

Comment:
nit: `click` is a more conventional word to describe button-like interactions. `press` is used more when you can do a `release` afterwards (e.g. a key on a keyboard).

```suggestion
    private ObjectProperty<EventHandler<RecalculateEvent>> propertyOnRecalculateClicked = new SimpleObjectProperty<>();
```

If you change this, also make sure to rename all the other instances